sway-parse/src/module.rs

```diff
@@ -23,16 +30,50 @@ impl Parse for ModuleKind {
     }
 }
 
-impl ParseToEnd for Module {
+impl ParseToEnd for Annotated<Module> {
     fn parse_to_end<'a, 'e>(mut parser: Parser<'a, '_>) -> ParseResult<(Self, ParserConsumed<'a>)> {
+        // Parse the attribute list.
+        let mut attribute_list = Vec::new();
+        while let Some(DocComment { .. }) = parser.peek() {
+            let doc_comment = parser.parse::<DocComment>()?;
+            // TODO: Use a Literal instead of an Ident when Attribute args
+            // start supporting them and remove `Ident::new_no_trim`.
+            let value = Ident::new_no_trim(doc_comment.content_span.clone());
+            let hash_kind = match &doc_comment.doc_style {
+                DocStyle::Inner => Ok(AttributeHashKind::Inner(HashBangToken::new(
+                    doc_comment.span.clone(),
+                ))),
+                DocStyle::Outer => Err(parser.emit_error(ParseErrorKind::ExpectedModuleDocComment)),
+            }
+            .unwrap();
```

Comment:
Do we want to `unwrap` here? Or should we return an error somehow? (Not exactly sure how `ParseResult` works off the top of my head).

Reply:
I'll look into it. I was using `Result` in the heat of the moment as a wrapper type.